src/Service.Tests/GraphQLBuilder/Sql/StoredProcedureBuilderTests.cs

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Azure.DataApiBuilder.Auth;
+using Azure.DataApiBuilder.Config;
+using Azure.DataApiBuilder.Service.GraphQLBuilder.Mutations;
+using Azure.DataApiBuilder.Service.GraphQLBuilder.Queries;
+using Azure.DataApiBuilder.Service.GraphQLBuilder.Sql;
+using Azure.DataApiBuilder.Service.Tests.GraphQLBuilder.Helpers;
+using HotChocolate.Language;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using static Azure.DataApiBuilder.Service.GraphQLBuilder.GraphQLTypes.SupportedTypes;
+
+namespace Azure.DataApiBuilder.Service.Tests.GraphQLBuilder.Sql
+{
+    /// <summary>
+    /// Validate GraphQL schema creation for stored procedure entities.
+    /// </summary>
+    [TestClass]
+    public class StoredProcedureBuilderTests
+    {
+        private Dictionary<string, EntityMetadata> _entityPermissions;
+
+        /// <summary>
+        /// Validates that a default value for stored procedure parameter defined in the runtime config
+        /// casts to the value type defined in the database schema because inferring the degree of precision
+        /// and associated value type of a numeric value written in the config is not sufficient.
+        /// - byte[] example referenced from Microsoft Docs example of Convert.FromBase64String(String) because
+        /// a byte array would be represented in a JSON payload as a string.
+        /// </summary>
+        /// <seealso cref="https://learn.microsoft.com/dotnet/api/system.convert.frombase64string?view=net-6.0#examples"/>
+        /// <seealso cref="https://learn.microsoft.com/dotnet/csharp/language-reference/builtin-types/floating-point-numeric-types"/>
+        /// <param name="systemType">Denotes system value type of stored procedure parameter.</param>
+        /// <param name="expectedGraphQLType">Target GraphQL type of parameter</param>
+        /// <param name="configParamValue">Explicit parameter default value set in runtime configuration.</param>
+        [DataTestMethod]
+        [DataRow(typeof(byte), BYTE_TYPE, 64, DisplayName = "Byte")]
+        [DataRow(typeof(short), SHORT_TYPE, 32767, DisplayName = "Short")]
+        [DataRow(typeof(int), INT_TYPE, 2147483647, DisplayName = "Int")]
+        [DataRow(typeof(long), LONG_TYPE, 9223372036854775807, DisplayName = "Long")]
+        [DataRow(typeof(float), SINGLE_TYPE, 3.402823e38, DisplayName = "Single")]
+        [DataRow(typeof(double), FLOAT_TYPE, "12.7", DisplayName = "Float")]
+        [DataRow(typeof(decimal), DECIMAL_TYPE, "12.7", DisplayName = "Decimal")]
+        [DataRow(typeof(string), STRING_TYPE, "paramValueconfig", DisplayName = "String")]
+        [DataRow(typeof(bool), BOOLEAN_TYPE, true, DisplayName = "Bool")]
+        [DataRow(typeof(DateTime), DATETIME_TYPE, "12/31/2030 12:00:00 AM", DisplayName = "DateTime")]
```

Comment:
could you please add DateTimeOffset too here?

Reply:
added. This did require a tweak to some of the code from your datetimeoffset pr. We have to distinguish between DateTime and DateTimeOffset, and the GraphQL hotchocolate type allows us to distinguish between them.

Reply:
Could you please point me to the tweak that you mention here? Or is there no more tweak required after our sync?

Reply:
no tweak needed anymore